### DISCUSSION TITLE
Core should not contain any hardcoded module names

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#33427] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Is your feature request related to a problem?

`$moduleManager->isInstalled('ps_emailsubscription')`
https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/admin/AdminStatsController.php#L790

✅ Removed - `$moduleManager->isInstalled('ps_categorytree')`
https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/front/ProductController.php#L243

### Describe the solution you'd like

We should not be dependent on any hardcoded module names.

### Do you plan to work on this feature?

- [x] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [x] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Seb33300** [comment]
PrestaShop is not depending on them.
`isInstalled` method check if the module is installed to prevent PrestaShop to fail if the module is not here.

**Comment by Hlavtox** [comment]
Yes, but why should `$this->context->cookie->last_visited_category = (int) $this->category->id_category;` be available only for the default category module? It should be universal, so all modules have same oportunities.

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon.
Waiting for your PR :rocket: 

Thank you

**Comment by kpodemski** [comment]
@Hlavtox if I understand correctly, newsletter should hook into the stats dashboard for the core to be clean, how does that sound?
